logstash-core/lib/logstash/runner.rb

```diff
@@ -482,8 +482,17 @@ def execute
 
   def running_as_superuser
     if Process.euid() == 0
+      unless @settings.set?("allow_superuser")
+        deprecation_logger.deprecated("WARNING: You are currently running Logstash with superuser privileges. " +
+          "Starting from version 9, this will be disabled by default. " +
+          "To avoid disruption during the upgrade, set 'allow_superuser' to true now if you wish to continue running as superuser temporarily after the upgrade. " +
+          "Note that this is not recommended for security reasons.")
```

Comment:
I would a bit simplify the statement.
```suggestion
        deprecation_logger.deprecated("WARNING: You are currently running Logstash with superuser privileges which is not recommended for security reasons. " +
          "Starting from version 9, this will not be allowed by default unless 'allow_superuser' is intentionally set to true.")
```

Reply:
@karenzone could you suggest how to do the deprecation communication? We want to suggest user to set 'allow_superuser' to true now for preparing v9 upgrade. The setting is optional at this moment, I am not sure how to deliver the message.

Reply:
Thanks for the ping and clarification, @kaisecheng and @robbavey. I'll take a look.

Reply:
Thanks for the question and the additional clarification.  It seems like @mashhurs is on the right track with his suggestion.  I started with his suggestion and iterated on it.  

Does this achieve your goals:

"Starting from version 9, running with superuser privileges is not permitted unless you explicitly set 'allow_superuser' to `true`, thereby acknowledging the possible security risks."

Reply:
Thanks for the question and the additional clarification. It seems like @mashhurs is on the right track with his suggestion. I started with his suggestion and iterated on it.

Does this achieve your goals:

"Starting from version 9, running with superuser privileges is not permitted unless you explicitly set 'allow_superuser' to true, thereby acknowledging the possible security risks."

If not, please let me know and we'll get it right.

Reply:
The majority of users do not aware the recent added `allow_superuser` setting and likely don't know the value is set to `true`. When users run the default 8.x Logstash, they will see the following

Starting from version 9, running with superuser privileges is not permitted unless you explicitly set 'allow_superuser' to true, thereby acknowledging the possible security risks.
NOTICE: Allowing Logstash to run as superuser is heavily discouraged as it poses a security risk. It is strongly recommended to set 'allow_superuser' to false.

As a user, I take the first line as a general announcement that it advices to set a setting to true if Logstash is run by root user. It doesn't tell me if I am affected. The second line advices me to set the same setting to false. It seems giving me two directions. 

I think overall it is fine to have two instruction. 
The first line was printed when the setting is unset, meaning the default (true) is used , while the second line is for the setting explicitly set to true to say it is better to set to false for security

I think we need to at least update the first line to inform user you are affected.

Reply:
I like where you're going with this.  Some minor wording tweaks for your consideration: 

Starting from version 9.0, running with superuser privileges is not permitted unless you explicitly set 'allow_superuser' to true, thereby acknowledging the possible security risks.
NOTICE:  Running Logstash as a superuser is strongly discouraged as it poses a security risk. 
Set 'allow_superuser' to `false` for better security.

Please let me know what you think.  From here, we can iterate quickly if necessary so that you can get this one over the finish line.